### PR TITLE
Add the fixed client-level subscriptionId parameter for ARM

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/2348

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -112,7 +112,7 @@
     {
       "name": "[TypeSpec] Smoke Test Debug",
       "request": "launch",
-      "cwd": "${workspaceFolder}/packages/typespec-test/test/customWrapper",
+      "cwd": "${workspaceFolder}/packages/typespec-test/test/NetworkAnalytics.Management",
```
Comment (MaryGao) [inline .vscode/launch.json:115]:
```suggestion
      "cwd": "${workspaceFolder}/packages/typespec-test/test/customWrapper",

```

File: .vscode/launch.json [hunk 2]
```diff
@@ -113,6 +113,7 @@
       "name": "[TypeSpec] Smoke Test Debug",
       "request": "launch",
       "cwd": "${workspaceFolder}/packages/typespec-test/test/customWrapper",
+
```
Comment (MaryGao) [inline .vscode/launch.json:116]:
```suggestion

```

File: packages/typespec-ts/src/modular/helpers/classicalOperationHelpers.ts [hunk 3]
```diff
@@ -9,13 +9,26 @@ import {
...
+import { SdkContext } from "../../utils/interfaces.js";
 
+export function shouldPromoteSubId(
+  dpgContext: SdkContext,
+  operationGroup: OperationGroup
+) {
+  const hasSubId = operationGroup.operations.some((op) =>
+    op.parameters.some((p) => p.clientName === "subscriptionId")
+  );
+  return dpgContext?.rlcOptions?.azureArm && hasSubId;
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/classicalOperationHelpers.ts:21]: Always require this logic when azureArm = true.

File: packages/typespec-ts/src/modular/buildClassicalClient.ts [hunk 4]
```diff
@@ -207,6 +210,8 @@ function buildClientOperationGroups(
       operationGroup.namespaceHierarchies[0] ?? operationGroup.propertyName,
       NameType.Property
     );
+    // TODO: remove this logic once client-level parameter design is finalized
```
Comment (joheredi) [inline packages/typespec-ts/src/modular/buildClassicalClient.ts:213]: can we link to a GH issue?

File: packages/typespec-ts/src/modular/buildClassicalClient.ts [hunk 5]
```diff
@@ -207,6 +210,8 @@ function buildClientOperationGroups(
...
+    const hasSubIdPromoted = shouldPromoteSubId(dpgContext, operationGroup);
```
Comment (joheredi) [inline packages/typespec-ts/src/modular/buildClassicalClient.ts:214]: lets use the full name for clarity `hasSubscriptionIdPromoted` and `shouldPromoteSubscriptionId`

File: packages/typespec-ts/src/modular/helpers/classicalOperationHelpers.ts [hunk 6]
```diff
@@ -9,13 +9,26 @@ import {
 import { Client, OperationGroup } from "../modularCodeModel.js";
 import { getClassicalLayerPrefix, getClientName } from "./namingHelpers.js";
 import { getOperationFunction } from "./operationHelpers.js";
+import { SdkContext } from "../../utils/interfaces.js";
 
+export function shouldPromoteSubId(
+  dpgContext: SdkContext,
+  operationGroup: OperationGroup
+) {
+  const hasSubId = operationGroup.operations.some((op) =>
```
Comment (joheredi) [inline packages/typespec-ts/src/modular/helpers/classicalOperationHelpers.ts:18]: lets use the whole subscription word, I'd rather make it clear than short

**Review by joheredi — APPROVED** [review]

File: packages/typespec-ts/src/modular/buildClassicalClient.ts (hunk 5 above)
Comment (MaryGao) [reply packages/typespec-ts/src/modular/buildClassicalClient.ts]: done

File: packages/typespec-ts/src/modular/helpers/classicalOperationHelpers.ts (hunk 6 above)
Comment (MaryGao) [reply packages/typespec-ts/src/modular/helpers/classicalOperationHelpers.ts]: updated.

**Pull request merged**